### PR TITLE
Update python code and tests for ros2

### PR DESCRIPTION
**Pull request by jonbinney**
Tried to make the fewest possible changes.

### CODE REVIEW

File: test/projection_test.py [hunk 1]
```diff
@@ -26,17 +28,17 @@ def build_constant_scan(
...
     scan.angle_max = angle_max
     scan.angle_increment = angle_increment
-    scan.scan_time = scan_time.to_sec()
+    scan.scan_time = scan_time.nanoseconds / 1e9
     scan.range_min = PROJECTION_TEST_RANGE_MIN
     scan.range_max = PROJECTION_TEST_RANGE_MAX
     scan.ranges = [range_val for _ in range(count)]
     scan.intensities = [intensity_val for _ in range(count)]
-    scan.time_increment = scan_time.to_sec()/count
+    scan.time_increment = scan_time.nanoseconds / 1e9 / count
```
Comment (jonbinney) [inline test/projection_test.py:39]: Manually dividing by `1e9` seems a bit hacky, but I don't think that ros2 Durations have any `to_sec()` equivalent.

**Comment by jonbinney** [comment]
Looks like CI failed; here is the error from the logs:
```
13:10:10 WARNING: At line 1 of file:/home/jenkins-agent/workspace/Rpr__laser_geometry__ubuntu_focal_amd64/ws/test_results/laser_geometry/projection.xunit.xml:cvc-complex-type.3.2.2: Attribute 'skip' is not allowed to appear in element 'testsuite'.
13:10:10 ERROR: Step ‘Publish xUnit test result report’ failed: The result file '/home/jenkins-agent/workspace/Rpr__laser_geometry__ubuntu_focal_amd64/ws/test_results/laser_geometry/projection.xunit.xml' for the metric 'JUnit' is not valid. The result file has been skipped.
```

This doesn't seem related to laser geometry.... maybe a problem with ament_cmake_nose? I don't have this problem locally when testing with foxy or galactic.

**Comment by jonbinney** [comment]
Looks like the xUnit jenkins plugin and nose disagree on whether the output field in the xml should be "skip" or "skipped". I'm tempted to just switch over to pytest.

**Comment by jonbinney** [comment]
I've switched the tests from nose to pytest, and now they pass on jenkins. Ready for review.

**Comment by jonbinney** [comment]
@mabelzhang gentle bump - this should be relatively low-risk to merge, but quite useful. Before this PR the python code in this package is completely broken on ROS2.

**Comment by mabelzhang** [comment]
Thanks for the PR and the tag!

I haven't set up CI in a while but if I set it up correctly and the test passes, we should be green...

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/15015/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/9754/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/12704/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/15177/)

**Comment by mabelzhang** [comment]
Do you need a release?

**Comment by jonbinney** [comment]
No release needed for now - thanks @mabelzhang !

**Pull request merged**